reactfront/src/components/graph-utils.js

```diff
@@ -110,3 +110,27 @@ export function boundIntervalKeepingLength([a, b], [min, max]) {
   else if (b > max) return [max - (b - a), max]
   else return [a, b]
 }
+
+// intended use:
+//   when indexing into an object that may or may not be undefined (using ?.)
+//   replace that call with this callback and it won't error out and just return undefined
+//   for example:
+//      undefOnBadRef(() => svg.current.getBoundingClientRect().left)
+//      replaces svgRef.current?.getBoundingClientRect().left
+// issues:
+//   this is more general and less precise with the error handling. If you need something to 
+//   only error on a bad index, do it directly (with x === null || x === undefined).
+export function undefOnBadRef (callback) {
+    let out
+    try {
+        out = callback()
+    } catch (err) {
+        if (err.name === ReferenceError) {
+            out = undefined
+        } else {
+            throw err
+        }
+    } finally {
+        return out
+    }
```

Comment:
This is a cool concept! Just a few issues:

I don't think that the exception handling works as it's supposed to though:
* Have you tested that `err.name === ReferenceError` works as expected? When I try getting a property of an undefined object in my JS console I get a `TypeError`, and it looks like the name is a string. `ReferenceError` looks like it's thrown when an undefined variable is accessed (eg `var a = undefinedVariable`) but it looks like `TypeError` is thrown when an undefined *property* of an existing variable/expression is accessed (eg `var a = b.nonExistentProperty.something`). I think a better condition is `err instanceof TypeError` or `err.name === "TypeError"`
* I'm not entirely sure why, but it seems like this catches *all* errors and doesn't actually throw other kinds of errors (even though your code looks like it should). For instance, `undefOnBadRef(() => decodeURI("%"))` should throw a `URIError` but it's silenced and returns `undefined`